pkg/apis/core/validation/validation.go

```diff
@@ -3608,6 +3675,32 @@ func validatePodConditions(conditions []core.PodCondition, fldPath *field.Path)
 	return allErrs
 }
 
+// ValidatePodEphemeralContainersUpdate tests that a user update to EphemeralContainers is valid.
+// newPod and oldPod must only differ in their EphemeralContainers.
+func ValidatePodEphemeralContainersUpdate(newPod, oldPod *core.Pod) field.ErrorList {
+	spec := newPod.Spec
+	specPath := field.NewPath("spec").Child("ephemeralContainers")
+
+	vols := make(map[string]core.VolumeSource)
+	for _, vol := range spec.Volumes {
+		vols[vol.Name] = vol.VolumeSource
+	}
+	allErrs := validateEphemeralContainers(spec.EphemeralContainers, spec.Containers, spec.InitContainers, vols, specPath)
+
+	if len(newPod.Spec.EphemeralContainers) < len(oldPod.Spec.EphemeralContainers) {
+		allErrs = append(allErrs, field.Forbidden(specPath, "Ephemeral Containers may not be removed"))
```

Comment:
don't capitalize this

Reply:
This was changed by a later commit, but I checked that all error strings are lower cased in the final product.

As I go through the rest of the comments I'll start switching "Ephemeral Container" to 'ephemeral container' in comments and docs as well.